packages/react-admin-core/src/table/useTableQueryFilter.tsx

```diff
@@ -0,0 +1,19 @@
+import { AnyObject } from "final-form";
+import * as React from "react";
+
+export interface IFilterApi<FilterValues extends AnyObject> {
+    current: FilterValues;
+    changeFilters: (values: FilterValues) => void;
+}
+export function useTableQueryFilter<FilterValues extends AnyObject>(defaultValues: FilterValues): IFilterApi<FilterValues> {
+    const [filters, setFilters] = React.useState<FilterValues>(defaultValues);
+
+    function changeFilters(v: FilterValues) {
+        setFilters(v);
```

Comment:
how does this object-state handle changes? I thought complex states should be implemented with useReducer? maybe this doesn't count as complex yet. https://reactjs.org/docs/hooks-reference.html#usereducer 

> useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values...

Reply:
I believe `useReducer` should be used once there are multiple different actions, for instance CRUD. As the only action for the filters would be to change them, using `useState` is perfectly fine in my opinion.

Reply:
Nothing is wrong with using useState here